straxen/plugins/event_patternfit.py

```diff
@@ -305,7 +316,42 @@ def compute_s2_llhvalue(self, events, result):
                     store_2LLH_ch = np.zeros((norm_llh_val.shape[0], self.config['n_top_pmts']) )
                     store_2LLH_ch[:, self.pmtbool_top] = norm_llh_val
                     result[t_+'_2llh_per_channel'][cur_s2_bool] = store_2LLH_ch
-                    
+
+    def compute_s2_neural_llhvalue(self, events, result):
+
+        downloader = straxen.MongoDownloader()
+        self.model_file = downloader.download_single(self.config['s2_tf_model'])
+        with tempfile.TemporaryDirectory() as tmpdirname:
+            tar = tarfile.open(self.model_file, mode="r:gz")
+            tar.extractall(path=tmpdirname)
+
+            import tensorflow as tf
+            def _logl_loss(patterns_true, likelihood):
+                return likelihood / 10.
+            self.model = tf.keras.models.load_model(tmpdirname,
+                                                    custom_objects={"_logl_loss": _logl_loss})
+            self.model_chi2 = tf.keras.Model(self.model.inputs,
+                                             self.model.get_layer('Likelihood').output)
```

Comment:
I think rather than getting the file during run time we should do it only once during the setup of the plugin like it is done for the peak position reconstruction: https://github.com/XENONnT/straxen/blob/b73bd26feef431be55cf963e19a3d204739e1054/straxen/plugins/position_reconstruction.py#L52
Or am I missing something why we have to redo it for each chunk of data?

Reply:
Thank you for pointing this out, Daniel! I'll make this right in the next commit.